test/unit/test_semver.py

```diff
@@ -134,3 +136,15 @@ def test__resolve_to_specific_version(self):
                 create_range(None, '<=0.0.5'),
                 ['0.0.3', '0.1.4', '0.0.5']),
             '0.0.5')
+
+        self.assertEqual(
+            resolve_to_specific_version(
+                create_range('=0.4.5a2', '=0.4.5a2'),
+                ['0.4.5a1', '0.4.5a2']),
+            '0.4.5a2')
+
+        self.assertEqual(
```

Comment:
Testing the specific case raised by @lbk-fishtown in https://github.com/dbt-labs/dbt/issues/3578